VariantShredding.md

```diff
@@ -25,290 +25,316 @@
-By the same token, `variant_value` may be absent, which is equivalent to their value being always null (in which case the field will always have the value Null or have the type of the `typed_value` column).
+| Variant Type                | Equivalent Parquet Type      |
+|-----------------------------|------------------------------|
+| boolean                     | BOOLEAN                      |
+| int8                        | INT(8, signed=true)          |
+| int16                       | INT(16, signed=true)         |
+| int32                       | INT32 / INT(32, signed=true) |
+| int64                       | INT64 / INT(64, signed=true) |
+| float                       | FLOAT                        |
+| double                      | DOUBLE                       |
+| decimal4                    | DECIMAL(precision, scale)    |
+| decimal8                    | DECIMAL(precision, scale)    |
+| decimal16                   | DECIMAL(precision, scale)    |
+| date                        | DATE                         |
+| timestamp                   | TIMESTAMP(true, MICROS)      |
+| timestamp without time zone | TIMESTAMP(false, MICROS)     |
+| binary                      | BINARY                       |
+| string                      | STRING                       |
+| array                       | LIST; see Arrays below       |
+| object                      | GROUP; see Objects below     |
 
-# Unshredded values
+#### Primitive Types
 
-If all values can be represented at a given level by whichever of `object`, `array`, or `typed_value` is present, `variant_value` is set to null.
+Primitive values can be shredded using the equivalent Parquet primitive type from the table above for `typed_value`.
 
-If a value cannot be represented by whichever of `object`, `array`, or `typed_value` is present in the schema, then it is stored in `variant_value`, and the other fields are set to null.
-In the Parquet example above, if field `a` was an object or array, or a non-integer scalar, it would be stored in `variant_value`.
+Unless the value is shredded as an object (see [Objects](#objects)), `typed_value` or `value` (but not both) must be non-null.
 
-If a value is an object, and the `object` field is present but does not contain all of the fields in the value, then any remaining fields are stored in an object in `variant_value`.
-In the Parquet example above, if field `b` was an object of the form `{"c": 1, "d": 2}"`, then the object `{"d": 2}` would be stored in `variant_value`, and the `c` field would be shredded recursively under `object.c`.
+#### Arrays
 
-Note that an array is always fully shredded if there is an `array` field, so the above consideration for `object` is not relevant for arrays: only one of `array` or `variant_value` may be non-null at a given level.
+Arrays can be shredded using a 3-level Parquet list for `typed_value`.
```

Comment:
must be?

Reply:
I changed this to "Arrays can be shredded *by* using a 3-level Parquet list . . .` so it is clear that arrays are optionally shredded but when they are they use a 3-level Parquet list.